lib/apiservers/engine/backends/volume.go

```diff
@@ -96,12 +105,103 @@ func (v *Volume) Volumes(filter string) ([]*types.Volume, []string, error) {
 		log.Infof("%s", vol.Name)
 		volumeMetadata, err := extractDockerMetadata(vol.Metadata)
 		if err != nil {
-			return nil, nil, fmt.Errorf("error unmarshalling docker metadata: %s", err)
+			return nil, nil, derr.NewErrorWithStatusCode(fmt.Errorf("error unmarshalling docker metadata: %s", err), http.StatusInternalServerError)
 		}
 		volume := NewVolumeModel(vol, volumeMetadata.Labels)
 		volumes = append(volumes, volume)
 	}
-	return volumes, nil, nil
+
+	// Parse and validate filter args
+	volumeFilters, err := filters.FromParam(filter)
+	if err != nil {
+		return nil, nil, derr.NewErrorWithStatusCode(err, http.StatusInternalServerError)
```

Comment:
This error instance is created more than once.  Lets make it a const or var and return that instead.